### PR TITLE
fix: handle managed identity ClientID or ResourceID in acr generator

### PR DESCRIPTION
**Pull request by bonddim**
## Problem Statement
Trying to use ACR generator with kubelet managed identity as described in [docs ](https://external-secrets.io/latest/api/generator/acr)

```yaml
apiVersion: generators.external-secrets.io/v1alpha1
kind: ACRAccessToken
metadata:
  name: my-azurecr
spec:
  tenantId: 11111111-2222-3333-4444-111111111111
  registry: example.azurecr.io
  environmentType: "PublicCloud"
  auth:
    managedIdentity:
      identityId: "xxxxx" # Supposed to be managed identity client ID as in Azure Key Vault provider
```

Got such in logs
```log
ManagedIdentityCredential authentication failed. the requested identity isn't assigned to this resource\nGET http://169.254.169.254/metadata/identity/oauth2/token
```

## Proposed Changes

Check if the `identityID` contains a "/" and set the `ManagedIdentityCredentialOptions` accordingly, using `ResourceID` or `ClientID` as appropriate.

## Checklist

- [x] I have read the [contribution guidelines](https://external-secrets.io/latest/contributing/process/#submitting-a-pull-request)
- [x] All commits are signed with `git commit --signoff`
- [x] My changes have reasonable test coverage
- [x] All tests pass with `make test`
- [x] I ensured my PR is ready for review with `make reviewable`

### CODE REVIEW

**Comment by Skarlso** [comment]
I'm a little bit skeptical. Why hasn't this come up yet? I'm pretty sure people were using this generator already. 🤔

**Comment by bonddim** [comment]
I was surprised too, that no issues opened with this. 
Will update PR with unbreakable solution in few hours.

**Comment by Skarlso** [comment]
Thanks. :D It's always a surprise finding something that didn't actually work, but has been used. :D :D

**Comment by gusfcarvalho** [comment]
out of curiosity @bonddim do you need to have the kubelet identity to auth to keyvault?

IIRC this is typically unsafe (any pods on that node _can_ use this path to get any secrets they want out of your vault)

**Comment by gusfcarvalho** [comment]
(also FYI - managed-identity use case was originally for the deprecated aad-pod-identity feature https://azure.github.io/aad-pod-identity/docs/)

**Comment by bonddim** [comment]
> out of curiosity @bonddim do you need to use the kubelet identity to auth to keyvault?
> 
> IIRC this is typically unsafe (any pods on that node _can_ use this path to get any secrets they want out of your vault)
> 
>     * What I mean is that clever users (well intended or not) can abuse misconfiguration on your cluster to use this resource ID to fetch keyvault.

Probably, you didn't notice, but this fix is not for accessing KeyVault, but for ACR generator only. 
Kubelet identity on AKS should already have "AcrPull" permissions on integrated registry https://learn.microsoft.com/en-us/azure/aks/cluster-container-registry-integration. 
Why do I need to provide another identity for that? 


> (also FYI - managed-identity use case was originally for the deprecated aad-pod-identity feature https://azure.github.io/aad-pod-identity/docs/)

How is this relates to the fix?

**Comment by gusfcarvalho** [comment]
> Why do I need to provide another identity for that?

Fair point 😄 


> How is this relates to the fix?

It was to @Skarlso, trying to respond "how this feature worked" 😄

**Comment by gusfcarvalho** [comment]
hi @bonddim - can you just add a comment in ACR docs saying managedIdentity can be used to authz as the kubelet? 🙏

**Review by gusfcarvalho — APPROVED** [review]

**Comment by gusfcarvalho** [comment]
Thanks @bonddim ! Just waiting for CI to run but this already LGTM :)

**Pull request merged**
